runtime/src/main/java/org/corfudb/runtime/clients/LogUnitClient.java

```diff
@@ -90,7 +94,9 @@ public Integer getPort() {
         wr.setRank(rank);
         wr.setBackpointerMap(backpointerMap);
         wr.setGlobalAddress(address);
-        CompletableFuture<Boolean> cf = sendMessageWithFuture(CorfuMsgType.WRITE.payloadMsg(wr));
+        CompletableFuture<Boolean> cf = sendRequestWithFuture(
+                getWriteLogRequestMsg((LogData) wr.getData()), false, false);
+
```

Comment:
A `WriteRequest` is constructed, but its not being used when the message is sent. I would recommend eliminating that type and directly using the LogData object.

Reply:
Done.